spec/subnets-lifecycle.md

```diff
@@ -0,0 +1,68 @@
+
+Subnets within a single hierarchy tree have native communication protocols and are able to transfer assets and state without a custom bridge.
+
+## **Lifecycle**
+
+The lifecycle of a subnet begins when it’s deployed and ends when the subnet is closed.
+
+Before a subnet is created, one must specify the validator power allocation mode, which is called the `PermissionMode`. There are three kinds of permission mode at the moment:
+
+- `Collateral`: This means the power of the validator comes from the collateral staked. New validators can `join` the subnet, also `stake` more collateral, `unstake` collateral and finally `leave` the subnet
+- `Federated`: The power of the validator is set by the owner of the subnet
+- `Static`: The power of the validator is set when the subnet is created. This mode is used mainly for debugging and testing
+
+At the time of subnet creation, a minimum validator count requirement is set by the subnet creator.  If the subnet’s permission mode is collateral, the creator could also set the minimum collateral requirement for the subnet to be bootstrapped. A standard fee for the transaction on the parent network will be paid for the transaction that establishes the subnet.
+
+Before the minimal requirements are met, the subnet is in a `preBootstrap` state, once those conditions are met, the subnet is in a `postBootstap` state. Any operations performed on the subnet in `preBootstrap` state is recorded in the subnet `genesis`.
+
+For normal users, they can perform:
+
+- Prefund: Provide genesis balance to the subnet
+- Fund: Send fund from the parent to an address in the child subnet. The parent will track the total circulating supply of the subnet.
+- Release: Send fund from the child subnet to an address in the parent subnet
+- Cross Message Call: Call another contract in another subnet
+
+For validators in `Collateral` permission mode, they can:
+
+- Join: Put some stake into the subnet and becomes a validator
+- State: Add more stake into the subnet
+- Unstake: WIthdraw stake from the subnet
+- Leave: WIthdraw all stake from the subnet
+
+For detailed explanation on how `Collateral` and `FederatedPower` affect the validator’s voting power, see `SubnetValidation`.
+
+### Deployment
+
+For ipc powered subnet, it’s deployed through `SubnetRegistry`. This contract should be deployed together with the gateway. One can simply call [newSubnetActor](https://github.com/consensus-shipyard/ipc/blob/7af25c4c860f5ab828e8177927a0f8b6b7a7cc74/contracts/src/subnetregistry/RegisterSubnetFacet.sol#L22) method to deploy a new subnet under the gateway contract. But do note that this method requires permission. The creator of this registry could limit the access on who can deploy new subnets or there are no restrictions at all, see [line](https://github.com/consensus-shipyard/ipc/blob/7af25c4c860f5ab828e8177927a0f8b6b7a7cc74/contracts/src/subnetregistry/RegisterSubnetFacet.sol#L95).
```

Comment:
> For ipc powered subnet

Are there any other options? If yes, maybe mention that subnet abstracts away the implementation?

Reply:
currently we only have IPC implementation, but as long as the interface is aligned, it should be ok